schemachanger: remove regional-by-row fallback for CREATE INDEX

rafiss opened the pull request:
This is easy to support, since all the required functionality was added
when we added support for regional-by-row tables for ADD/DROP COLUMN.

Epic: CRDB-31329
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/139622)
<!-- Reviewable:end -->

rafiss committed:
schemachanger: remove regional-by-row fallback for CREATE INDEX

This is easy to support, since all the required functionality was added
when we added support for regional-by-row tables for ADD/DROP COLUMN.

Release note: None

fqazi reviewed (approved):
Nice! Just one suggestion for DML injection <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 16 of 16 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/139622#-:-OLk0CUM9grSq6nxBj2H:btsnwjc)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @rafiss)

___
*[`pkg/ccl/schemachangerccl/testdata/end_to_end/create_index_regional_by_row/create_index_regional_by_row.definition` line 9 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/139622#-OLk0--w1ey84Pn7Ul9T:-OLk0--w1ey84Pn7Ul9U:b-f1vg0f) ([raw file](https://github.com/cockroachdb/cockroach/blob/857bf2b3e6abbdb027673c9d1d484a4f609b6155/pkg/ccl/schemachangerccl/testdata/end_to_end/create_index_regional_by_row/create_index_regional_by_row.definition#L9)):*
> ```
> ----
> 
> stage-exec phase=PostCommitPhase stage=:
> ```

We probably also want inserts on the non-reveritble part of the plan


<!-- Sent from Reviewable.io -->

rafiss reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/139622)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @fqazi)

___
*[`pkg/ccl/schemachangerccl/testdata/end_to_end/create_index_regional_by_row/create_index_regional_by_row.definition` line 9 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/139622#-OLk0--w1ey84Pn7Ul9T:-OLkLoZzB4aW-yAto1UG:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/857bf2b3e6abbdb027673c9d1d484a4f609b6155/pkg/ccl/schemachangerccl/testdata/end_to_end/create_index_regional_by_row/create_index_regional_by_row.definition#L9)):*
<details><summary><i>Previously, fqazi (Faizan Qazi) wrote…</i></summary><blockquote>

We probably also want inserts on the non-reveritble part of the plan
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

rafiss commented:
tftr!

bors r+
